### PR TITLE
Add calculateRewards method in ChainDataProvider

### PR DESCRIPTION
**Pull request by courtneyeh**
#6701

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java [hunk 1]
```diff
@@ -388,6 +390,24 @@ public void getStateSyncCommittees_shouldRejectFarFutureEpoch() {
         .isCompletedExceptionallyWith(IllegalArgumentException.class);
   }
 
+  @Test
+  public void calculateRewards_shouldGetData() {
```
Comment (rolfyone) [inline data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java:394]: this is good, what happens when we ask for a validator index that's not in the committee? can we demonstrate that in a unit test?

**Review by rolfyone — APPROVED** [review]
LGTM, suggested another unit test to push an edge case.

File: data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java (hunk 1 above)
Comment (courtneyeh) [reply data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java:394]: Handled in next [PR](https://github.com/ConsenSys/teku/pull/6860)
